src/hooks/services/hooks.service.ts

```diff
@@ -169,7 +170,16 @@ export class HooksService {
 
       case AggregationDetailsMode.API:
         /** Get the JWT token */
-        const token = await this.aggregator.getJWToken(serviceAccountConfig, customer.aggregationDetails.userId);
+        let token: JWTokenResponse;
+        if (customer.aggregationDetails.userId !== undefined) {
+          token = await this.aggregator.getJWTokenForExistingUser(
+            serviceAccountConfig,
+            customer.aggregationDetails.userId,
+          );
+        } else {
+          token = await this.aggregator.getJWTokenForNewUser(serviceAccountConfig);
+        }
```

Comment:
I think you could use `const` here like this:
```suggestion
        const token: JWTokenResponse = customer.aggregationDetails.userId !== undefined
          ? await this.aggregator.getJWTokenForExistingUser(serviceAccountConfig, customer.aggregationDetails.userId)
          : await this.aggregator.getJWTokenForNewUser(serviceAccountConfig);
```